configs/xrootd-stashcache-origin-server.cfg

```diff
@@ -1,28 +1,47 @@
-# https://opensciencegrid.github.io/StashCache/
-
-xrd.allow host *
-sec.protocol  host
-sec.protbind  * none
-all.adminpath /var/spool/xrootd
-all.pidpath /var/run/xrootd
-
-# StashCache origin server settings
+#
+# Configure an origin server for the OSG Data Federation.
+#
+# **********************************************************************
+# * WARNING: DO NOT EDIT THIS FILE.  IT WILL BE OVERWRITTEN ON UPGRADE *
+# **********************************************************************
+#
+# Instead of editing this file,
+# - Edit `/etc/xrootd/config.d/10-origin-site-local.cfg` to provide a few
+#   MANDATORY variable values.
+# - Override OSG-provided defaults by dropping a site-specific configuration
+#   file in `/etc/xrootd/config.d` and prefixing it with "90-" or higher.
+#
+# This file is part of the StashCache Daemon
+# https://opensciencegrid.github.io/StashCache-Daemon
+#
 
 # The directory on local disk containing the files to share, e.g. "/stash".
-oss.localroot <local dir>
+oss.localroot $(origindir)
+
 # The directory under oss.localroot to share data from.
 # To share multiple subdirectories, have one `all.export` line per
-# subdirectory.
+# subdirectory; place additional exports in the config.d directory.
 #
 # NOTE: the StashCache namespace is global; do not share paths that may
 # collide with paths provided by other origin servers.
-all.export /<your VO>
+all.export $(originexport)
+
 
+# The location of data federation this XCache participates in
+#
+# By default, we set this to the OSG Data Federation
+all.manager redirector.osgstorage.org+ 1213
+#
+# Alternately, use the following line for a test sesrver
+#
+# all.manager redirector-itb.osgstorage.org+ 1213
+
+
+# A few reasonable defaults for the origin server.
 xrd.port 1094
 all.role server
-all.manager redirector.osgstorage.org+ 1213
 
-xrootd.trace emsg login stall redirect
-ofs.trace all
-xrd.trace all
-cms.trace all
+
+# Drop your site-specific configuration additions into this directory.
+continue /etc/xrootd/config.d
```

Comment:
In testing this PR, the daemon fails to start:
```181212 20:41:40 5603 XrootdStream: Variable originexport is undefined.
...
181212 20:41:40 5603 XrdProtocol: Protocol xrootd could not be loaded```

Looks like the `continue /etc/xrootd/config.d` needs to come before the `oss.localroot $(origindir)` and `all.export $(originexport)` entries so the variables are defined. How about moving those two blocks after `continue`?

Reply:
Unfortunately, `continue` is not `include`.  The `continue` statement stops further processing of the current configuration file.

So, we may need to move these directives to the `config.d` directory.

Reply:
If we set the `oss.localroot` and `all.export` parameters in `config.d/10-origin-site-local.cfg`, the service seems to run fine. I've not come up with a cleaner way yet.

Keep them as variables, so the service fails to start until after configuration?
```
# set origindir = /stash
oss.localroot $(origindir)

# set originexport = /VO
all.export $(originexport)
```